bpf/lib/maps.h

```diff
@@ -284,3 +285,13 @@ tail_call_internal(struct __ctx_buff *ctx, const __u32 index, __s8 *ext_err)
 	return DROP_MISSED_TAIL_CALL;
 }
 #endif /* SKIP_CALLS_MAP */
+
+#ifdef ENABLE_CPU_MAP
+struct {
+    __uint(type, BPF_MAP_TYPE_CPUMAP);
+    __type(key, __u32);
+    __type(value, struct bpf_cpumap_val);
+    __uint(pinning, LIBBPF_PIN_BY_NAME);
+    __uint(max_entries, __NR_ONLINE_CPUS__);
```

Comment:
Instead of leaning on the preprocessor for setting maxentries, I suggest we add a small case to ebpf-go to automatically set it when it's left at its default 0 value as well as clamping it to possible CPUs, like we do for PerfEventArray: https://github.com/cilium/ebpf/blob/9895aae6467e9f677846e5b1407893133dbbbfc9/map.go#L125.

Also, I'm not sure if limiting it to online CPUs makes sense, since the list can have holes and that would mean some CPUs get cordoned off, while others are wrongly included. For example, on a 16-core machine, if CPU 0 and 1 go offline, the cpumap would have 14 entries, wrongly including CPU 0 and 1 and leaving online CPUs 14 and 15 unutilized?

That said, I'm not sure if bpf can deal with offline CPUs here at all, since https://docs.kernel.org/bpf/map_cpumap.html only mentions bpf code using `bpf_num_possible_cpus()` and setting `libbpf_num_possible_cpus()` during map creation.

Reply:
Hey @ti-mo, I gave implementing setting the max entries a shot, see https://github.com/cilium/ebpf/pull/1627. It sets the max entries to the number of possible CPUs on the node. I'm thinking we can change the logic used to select a CPU index to account for online CPUs, rather than setting the max entries to online CPUs.